x-pack/plugins/apm/public/components/app/infra_overview/infra_tabs/index.tsx

```diff
@@ -0,0 +1,229 @@
+        return callApmApi('GET /internal/apm/services/{serviceName}/infra', {
+          params: {
+            path: { serviceName },
+            query: {
+              environment,
+              kuery,
+              start,
+              end,
+            },
+          },
+        });
+      }
+    },
+    [environment, kuery, serviceName, start, end]
+  );
+
+  const containerIds = data?.infrastructureData.containerIds;
+  const podNames = data?.infrastructureData.podNames;
+  const hostNames = data?.infrastructureData.hostNames;
+
+  const { data: containersData, status: containersStatus } = useFetcher(
+    (callApmApi) => {
+      if (containerIds && containerIds.length > 0) {
+        return callApmApi('GET /internal/apm/container/host_names', {
+          params: {
+            query: {
+              start,
+              end,
+              containerIds: JSON.stringify(containerIds),
+            },
+          },
+        });
+      }
+    },
+    [containerIds, start, end]
+  );
+
+  const tabs = useTabs({
+    containerIds: containerIds || [],
+    podNames: podNames || [],
+    hostNames: hostNames || [],
+    containersHostNames: containersData?.hostNames || [],
+    start,
+    end,
+  });
+
+  if (
+    status === FETCH_STATUS.LOADING ||
+    containersStatus === FETCH_STATUS.LOADING
+  ) {
+    return (
+      <div style={{ textAlign: 'center' }}>
+        <EuiLoadingSpinner size="l" />
+      </div>
+    );
+  }
+
+  if (status === FETCH_STATUS.FAILURE) {
+    return (
+      <div style={{ textAlign: 'center' }}>
+        <FailurePrompt />
+      </div>
+    );
+  }
+
+  if (
+    status === FETCH_STATUS.SUCCESS &&
+    containerIds &&
+    containerIds.length <= 0 &&
+    podNames &&
+    podNames.length <= 0 &&
+    hostNames &&
+    hostNames?.length <= 0
+  ) {
+    return (
+      <div style={{ textAlign: 'center' }}>
+        <EmptyPrompt />
+      </div>
+    );
+  }
+
+  return (
+    <>
+      <EuiTabbedContent
+        tabs={tabs}
+        initialSelectedTab={tabs[0]}
+        autoFocus="selected"
+        onTabClick={(tab) => {}}
+      />
+    </>
+  );
+}
+
+function useTabs({
+  containerIds,
+  podNames,
+  hostNames,
+  containersHostNames,
+  start,
+  end,
+}: {
+  containerIds: string[];
+  podNames: string[];
+  hostNames: string[];
+  containersHostNames: string[];
+  start: string;
+  end: string;
+}) {
+  const { services } = useKibana<ApmPluginStartDeps>();
+  const { infra } = services;
+  const HostMetricsTable = infra?.HostMetricsTable;
+  const ContainerMetricsTable = infra?.ContainerMetricsTable;
+  const PodMetricsTable = infra?.PodMetricsTable;
+
+  const timerange = useMemo(
+    () => ({
+      from: start,
+      to: end,
+    }),
+    [start, end]
+  );
+
+  const hostsFilter = useMemo(
+    (): QueryDslQueryContainer => ({
+      bool: {
+        should: [
+          {
+            terms: {
+              'host.name':
+                containersHostNames.length > 0
```

Comment:
Is there a point to merge the lists instead of getting one or the other for cases where a service is both directly on a host and in a container. (unlikely but just thought about this)

Reply:
Right now, the host names we are getting from the apm documents, (first call to the api) if there's container id, they are actually pod names, so we don't want to rely on them if the service is containerized